devito/propagator.py

```diff
@@ -46,18 +73,25 @@ def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, op
         self.fd = FunctionDescriptor(name)
         self.pre_loop = []
         self.post_loop = []
+
         if self.profile:
             self.add_local_var("time", "double")
             self.pre_loop.append(cgen.Statement("struct timeval start, end"))
             self.pre_loop.append(cgen.Assign("time", 0))
             self.post_loop.append(cgen.PrintStatement("time: %f\n", "time"))
+
+        # Auto tuning
+        if self.cache_blocking and self.auto_tune:
+            self.at_markers = [("M1_start", "M1_end"), ("M2_start", "M2_end")]  # markers for at pragmas
```

Comment:
They could be set as class variables (NOT instance variables as here) in at_controller since that should be the class defining the at codebase.